### PR TITLE
Redesign "Select account to Join" modal when joining a community

### PR DESCRIPTION
**Pull request by salman-neslit**
## Link to Issue
Closes: #9555

## Description of Changes
- Added a visually appealing design for selecting addresses when joining a community, including radio buttons and a Join button.
- Updated the layout to align with the new design specs for improved user experience.

## "How We Fixed It"
- Implemented radio buttons for address selection and updated the styling to make the join process more engaging and user-friendly.

## Test Plan
- Unit tested the `FIXME()` call.
- CA (click around) tested on local and frack:
  - TODO page
  - 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Comment by salman-neslit** [comment]
https://github.com/user-attachments/assets/69afa683-592a-480c-9efb-3f8b0bfecc2f

**Comment by salman-neslit** [comment]
In this PR, I've implemented a visually appealing design for selecting addresses when joining a community, The layout has also been updated to align with new design specifications for an improved user experience.

This update also includes "Remove" and "Set Default" buttons, but their functionality needs clarification. It seems counterintuitive to use the join community modal for address removal, as this action would logically belong on the community page with a separate modal dedicated to managing addresses or leaving a community.

Regarding the "Set Default" functionality, I have some concerns that it could introduce issues within multi-chain, multi-community scenarios, potentially affecting certain branching cases. Additionally, if we decide to implement a "Set Default" address feature, we should determine whether the default should be stored per community or per user profile. If it’s per community, we should consider where and how this default address would be stored if it falls outside the community scope.

To unblock myself, I’ve taken a particular approach here, but I welcome any suggestions or alternative solutions.

File: packages/commonwealth/client/scripts/views/components/component_kit/AccountSelector/AccountSelector.tsx [hunk 1]
```diff
@@ -38,35 +40,16 @@ const LinkAccountItem = ({
...
-          <div className="account-user">
-            <User
-              userAddress={address}
-              userCommunityId={app.chain?.id || walletNetwork}
-              shouldHideAvatar
-            />
-          </div>
+    <div className="account-item account-item-emphasized">
+      <div className="account-item-left" onClick={() => onToggle(idx)}>
+        <input type="radio" checked={isChecked} />
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/AccountSelector/AccountSelector.tsx:49]: you should use CWRadioButton from new_designs directory

**Review by masvelio — APPROVED** [review]
please discuss your valid concerns with product so we can merge this as a first part and create ticket with enhancement if more details will be known. I think that we should stick to simple solution as it is now. As you correctly  noticed, removing and setting a default address does not seem to be a good fit here

**Comment by Israellund** [comment]
@sachben91 Can you address Salman and Marcin's comments above so that we can get this merged?

**Comment by sachben91** [comment]
lets not implement "set default" or "remove address" here. Lets close this out with the above change which allows the user to make an address selection before joining. 

For "removing address" @sssssabinaaa I'd suggest this can either be part of the right drop down where you see the addresses or we have another drop down on the left sidebar when you click on the address you have joined with

**Review by Israellund — APPROVED** [review]

**Review by mzparacha — APPROVED** [review]

**Pull request merged**